clients/roscpp/src/libros/transport/transport_tcp.cpp

```diff
@@ -179,7 +179,7 @@ void TransportTCP::setKeepAlive(bool use, uint32_t idle, uint32_t interval, uint
     }
 
 /* cygwin SOL_TCP does not seem to support TCP_KEEPIDLE, TCP_KEEPINTVL, TCP_KEEPCNT */
-#if defined(SOL_TCP) && !defined(__CYGWIN__)
+#if defined(SOL_TCP) && !defined(__CYGWIN__) && defined(TCP_KEEPIDLE) && defined(TCP_KEEPINTVL) && defined(TCP_KEEPCNT)
```

Comment:
Is the not-Cygwin check still required now that you're checking all these other defines?

Reply:
I don't know, could be that it's defined but shouldn't be used...

Reply:
Makes sense. It's frustrating not having any CI coverage other than the officially-supported platforms.

@dirk-thomas Is there a reason we don't use Travis and Appveyor to get some more coverage of this repo, particularly thinking in this instance of Windows and Python 3.

Reply:
> Is there a reason we don't use Travis and Appveyor to get some more coverage of this repo, particularly thinking in this instance of Windows and Python 3?

The only reason is that nobody has spend the time to setup something like that (and is committed to maintain that setup in the future :wink:).

Reply:
Oh, okay! I might be willing, as we're contemplating a Python 3 migration. I just wasn't sure if there was some conviction that we should be using only the buildfarm for testing or something.